src/main/java/frc/robot/subsystems/drive/TalonSwerveModule.java

```diff
@@ -0,0 +1,171 @@
+package frc.robot.subsystems.drive;
+
+import com.ctre.phoenix6.configs.CANcoderConfiguration;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.VelocityDutyCycle;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.AbsoluteSensorRangeValue;
+import com.ctre.phoenix6.signals.FeedbackSensorSourceValue;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import com.ctre.phoenix6.signals.SensorDirectionValue;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.util.sendable.Sendable;
+import edu.wpi.first.util.sendable.SendableBuilder;
+
+
+public class TalonSwerveModule implements Sendable{
+  public TalonFX drivingTalon;
+  public TalonFX turningTalon;
+  public CANcoder turningTalonEncoder;
+  public double turningRotationalOffset;
+
+  public SwerveModuleState desiredState = new SwerveModuleState(0.0, new Rotation2d());
+
+  public TalonSwerveModule(int drivingTalonCanId, int turningTalonCanId, int cancoderCanId, double turningRotationalOffsetValue){
+    drivingTalon = new TalonFX(drivingTalonCanId);
+    turningTalon = new TalonFX(turningTalonCanId);
+    turningTalonEncoder = new CANcoder(cancoderCanId);
+    turningRotationalOffset = turningRotationalOffsetValue;
+  }
```

Comment:
these lines were also in the constructor of the old drive code:

desiredState.angle = Rotation2d.fromRadians(turningRelativeEncoder.getPosition());
drivingTalon.setPosition(0);

i think the top line could be replaced with:
desiredState.angle = Rotation2d.fromRotations(turningTalon.getPosition().getValueAsDouble());

Reply:
done